posts/2020-05-27-how-to-analyze-open-liberty-logs-with-splunk.adoc

```diff
@@ -0,0 +1,139 @@
+
+image::/img/blog/blog_splunk_input_settings.png[Splunk Input Settings, align="center"]
+
+[start=3]
+. The HTTP Event Collector token is now created successfully.  Take note of the generated Token Value for later use.
+
+image::/img/blog/blog_splunk_token_created.png[Splunk Token Generated, align="center"]
+
+[start=4]
+. Enable access to the HTTP Event Collector, if it is disabled, by going to *Settings -> Data Inputs -> HTTP Event Collector -> Global Settings*, and click *Enabled* for the All Tokens field and check the *Enable SSL* checkbox. This will ensure the port 8088 for HTTP Event Collector is secured and is using the Splunk default certificate.
+
+image::/img/blog/blog_splunk_global_settings.png[Splunk Global Settings, align="center"]
+
+=== Running the Open Liberty Docker Image
+
+We’ll use the Open Liberty Docker image from Docker Hub. The official Docker GitHub repository for Open Liberty can be found link:https://github.com/OpenLiberty/ci.docker/[here]. You can follow instructions on how to build an application image, with your application, using the Open Liberty Docker image from https://github.com/OpenLiberty/ci.docker#building-an-application-image
+
+. Run the Open Liberty Docker image to start the Open Liberty server, with the following command.  This configures Open Liberty to emit JSON formatted logs and uses the Splunk logging driver with the HTTP Event Collector:
+[source]
+----
+   docker run -d -p 80:9080 -p 443:9443 --log-driver=splunk --log-opt splunk-url=https://localhost:8088 --log-opt splunk-token=<REPLACE_WITH_SPLUNK_HEC_TOKEN> --log-opt splunk-insecureskipverify=true --log-opt splunk-format=json -e WLP_LOGGING_CONSOLE_FORMAT=JSON -e WLP_LOGGING_CONSOLE_LOGLEVEL=info -e WLP_LOGGING_CONSOLE_SOURCE=message,trace,accessLog,ffdc,audit open-liberty:latest
+----
+
+Breakdown of the above command::
+
+* *--log-driver=splunk* : Configures the Open Liberty container to use the Splunk log driver.
+* *--log-opt splunk-url* : The host of your Splunk Enterprise Docker container, with the port and scheme of the configured HTTP Event Collector.
+* *--log-opt splunk-token* : The Splunk HTTP Event Collector token. Ensure that *<REPLACE_WITH_SPLUNK_HEC_TOKEN>* value from the above command is replaced by the generated HEC token from Step 3 from the “Configuring the HTTP Event Collector” section.
+* *--log-opt splunk-insecureskipverify=true* : Ignores the server certificate validation, as the Splunk Enterprise container instance is using a default self-signed certificate.
+* *--log-opt splunk-format* : Specifies the message format that will get sent to Splunk’s HTTP Event Collector.
+* *WLP_LOGGING_CONSOLE_FORMAT* : An environment variable that specifies the format for the Open Liberty runtime console.
+* *WLP_LOGGING_CONSOLE_LOGLEVEL* : An environment variable which is a filter that controls the granularity of messages that go to the console.
```

Comment:
**change**
An environment variable which is a filter that controls the granularity of messages that go to the console.
**to
An environment variable that is a filter that controls the granularity of messages that go to the console.